docs/docs/reference_home.md

```diff
@@ -1,3 +1,183 @@
+you want with `polars` by specifying the **data structure** you want to use and 
+then by applying **expressions** in a particular **context**.
 
-Here's the reference homepage.
+
+## Data structure
+
+As explained in some vignettes, one of `polars` biggest strengths is the ability
+to choose between eager and lazy evaluation, that require respectively a 
+`DataFrame` and a `LazyFrame` (with their counterparts `GroupBy` and `LazyGroupBy`
+for grouped data). 
+
+We can apply functions directly on a `DataFrame` or `LazyFrame`, such as `rename()`
+or `drop()`. Most (but not all!) functions that can be applied to `DataFrame`s
+can also be used on `LazyFrame`s. Calling `<DataFrame>$lazy()` yields
+a `LazyFrame`. While calling `<LazyFrame>$collect()` starts a computation and
+yields a `DataFrame` as result.
+
+Another common data structure is the `Series`, which can be considered as the 
+equivalent of R vectors in `polars`' world. Therefore, a `DataFrame` is a list of
+`Series`.
+
+Operations on `DataFrame` or `LazyFrame` are useful, but many more operations
+can be applied on columns themselves by using various **expressions** in different
+**contexts**.
+
+
+
+## Contexts
+
+A context simply is the type of data modification that is done. There are 3 types
+of contexts:
+
+* select and modify columns with `select()` and `with_columns()`;
+* filter rows with `filter()`;
+* group and aggregate rows with `groupby()` and `agg()`
+
+Inside each context, you can use various **expressions** (aka. `Expr`). Some expressions cannot
+be used in some contexts. For example, in `with_columns()`, you can only apply
+expressions that return either the same number of values or a single value that
+will be duplicated on all rows:
+
+```r
+test = pl$DataFrame(mtcars)
+
+# this works
+test$with_columns(
+  pl$col("mpg") + 1
+)
+
+# this doesn't work because it returns only 2 values, while mtcars has 32 rows.
+test$with_columns(
+  pl$col("mpg")$slice(0, 2)
+)
+```
+By contrast, in an `agg` context, any number of return values are possible, as
+they are returned in a list, and only the new columns or the grouping columns 
+are returned.
+
+```r
+test$groupby(pl$col("cyl"))$agg(
+  pl$col("mpg"), # varying number of values
+  pl$col("mpg")$slice(0, 2)$suffix("_sliced"), # two values
+  # aggregated to one value and implicitly unpacks list
+  pl$col("mpg")$sum()$suffix("_summed") 
+)
+
+shape: (3, 4)
+┌─────┬──────────────────────┬──────────────┬────────────┐
+│ cyl ┆ mpg                  ┆ mpg_sliced   ┆ mpg_summed │
+│ --- ┆ ---                  ┆ ---          ┆ ---        │
+│ f64 ┆ list[f64]            ┆ list[f64]    ┆ f64        │
+╞═════╪══════════════════════╪══════════════╪════════════╡
+│ 4.0 ┆ [22.8, 24.4, … 21.4] ┆ [22.8, 24.4] ┆ 293.3      │
+│ 8.0 ┆ [18.7, 14.3, … 15.0] ┆ [18.7, 14.3] ┆ 211.4      │
+│ 6.0 ┆ [21.0, 21.0, … 19.7] ┆ [21.0, 21.0] ┆ 138.2      │
+└─────┴──────────────────────┴──────────────┴────────────┘
```

Comment:
I think the R code and the results are mixed up and not very good.
Could this file be changed to Rmd?

I believe it is possible by adding a target building this file to the Makefile and gitignore this file.